Mathlib/Data/Nat/Basic.lean

```diff
@@ -0,0 +1,162 @@
+/-
+Copyright (c) 2014 Floris van Doorn (c) 2016 Microsoft Corporation. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Floris van Doorn, Leonardo de Moura, Jeremy Avigad, Mario Carneiro
+-/
+import Mathlib.Data.Nat.Init
+import Mathlib.Logic.Function.Basic
+import Mathlib.Logic.Nontrivial.Defs
+import Mathlib.Tactic.Contrapose
+import Mathlib.Tactic.GCongr.CoreAttrs
+import Mathlib.Tactic.PushNeg
+import Mathlib.Util.AssertExists
+
+/-!
+# Basic operations on the natural numbers
+
+This file builds on `Data.Nat.Init` by adding basic lemmas on natural numbers
+depending on Mathlib definitions.
+
+See note [foundational algebra order theory].
+-/
+
+/- We don't want to import the algebraic hierarchy in this file. -/
+assert_not_exists Monoid
+
+open Function
+
+namespace Nat
+variable {a b c d m n k : ℕ} {p : ℕ → Prop}
+
+-- TODO: Move the `LinearOrder ℕ` instance to `Order.Nat` (https://github.com/leanprover-community/mathlib4/pull/13092).
+instance instLinearOrder : LinearOrder ℕ where
+  le := Nat.le
+  le_refl := @Nat.le_refl
+  le_trans := @Nat.le_trans
+  le_antisymm := @Nat.le_antisymm
+  le_total := @Nat.le_total
+  lt := Nat.lt
+  lt_iff_le_not_le := @Nat.lt_iff_le_not_le
+  decidableLT := inferInstance
+  decidableLE := inferInstance
+  decidableEq := inferInstance
+
+-- Shortcut instances
+instance : Preorder ℕ := inferInstance
+instance : PartialOrder ℕ := inferInstance
+instance : Min ℕ := inferInstance
+instance : Max ℕ := inferInstance
+instance : Ord ℕ := inferInstance
+
+instance instNontrivial : Nontrivial ℕ := ⟨⟨0, 1, Nat.zero_ne_one⟩⟩
+
+attribute [gcongr] Nat.succ_le_succ Nat.div_le_div_right Nat.div_le_div_left Nat.div_le_div
+
+/-! ### `succ`, `pred` -/
+
+lemma succ_injective : Injective Nat.succ := @succ.inj
+
+lemma succ_ne_succ : succ m ≠ succ n ↔ m ≠ n := succ_injective.ne_iff
```

Comment:
Consider moving to Init

Reply:
I had to do a manual proof, that's what you meant right?